### DISCUSSION TITLE
Idea:  Add a curve control of weight ! Just like PhotoShop,  It matters final result

### DISCUSSION THREAD

**Discussion by LanceGao97** [lllyasviel/ControlNet#434] [Ideas]
Hello,  
I have been playing Stable diffusion for 3 months, and I always put my old sketches into Control net to give control of the final images, 
what I found is   , when I see the sample images morphing,  it's very clearly  that I can see the shape or color is controled by the line art or by the color,  or the prompt,        if I  let the guidance of CN to be 0.5  , then   at 15steps of 30,   the sample images will go out of control, 
some times this is needed  to make better rendering,  but sometimes this is not good,
So   can we make a curve control  just like curve in Photoshop?,   I think this function won't cost too many nights for scriptors, but this is real good for final image control,  
![11题](https://github.com/lllyasviel/ControlNet/assets/125782753/bdbae2a0-f2c6-4a1f-bc24-b7dffeabbbbc)

this image is what I meant,   some times you need to control weight to be 1 at first,  some other times you need it to be a nice curve.

**Comment by geroldmeisinger** [comment]
I think this a good idea but it has to be implement by your Stable Diffusion UI (Automatic1111). ControlNet itself supports individual strength per step already and most UIs provide paramaters such as guidance-start, start-percent etc.
